staging/src/k8s.io/kubectl/pkg/polymorphichelpers/history.go

```diff
@@ -101,24 +102,16 @@ type DeploymentHistoryViewer struct {
 // ViewHistory returns a revision-to-replicaset map as the revision history of a deployment
 // TODO: this should be a describer
 func (h *DeploymentHistoryViewer) ViewHistory(namespace, name string, revision int64) (string, error) {
-	versionedAppsClient := h.c.AppsV1()
-	deployment, err := versionedAppsClient.Deployments(namespace).Get(context.TODO(), name, metav1.GetOptions{})
-	if err != nil {
-		return "", fmt.Errorf("failed to retrieve deployment %s: %v", name, err)
-	}
-	_, allOldRSs, newRS, err := deploymentutil.GetAllReplicaSets(deployment, versionedAppsClient)
+	allRSs, err := getDeploymentReplicaSets(h.c.AppsV1(), namespace, name)
 	if err != nil {
-		return "", fmt.Errorf("failed to retrieve replica sets from deployment %s: %v", name, err)
-	}
-	allRSs := allOldRSs
-	if newRS != nil {
-		allRSs = append(allRSs, newRS)
+		return "", err
 	}
 
 	historyInfo := make(map[int64]*corev1.PodTemplateSpec)
 	for _, rs := range allRSs {
 		v, err := deploymentutil.Revision(rs)
 		if err != nil {
+			klog.Warningf("unable to get revision from replicaset %s for deployment %s in namespace %s", rs.Name, name, namespace)
```

Comment:
nit: would it be better to lower log level(ie. `klog.v(2)`) and embed error string into the message?

Reply:
Thanks.  I added the error to the message.

But I'm leaving it as a warning, for now at least.

I don't think there would be a normal situation where an error occurs while getting the replicaset revision, so my intention is that the caller be made aware of the problem by printing a warning message.  If this happens, then either there is something corrupt about that replicaset, or perhaps a bug in the revision logic somewhere in k8s that causes that situation (should it even occur).

Reply:
> I don't think there would be a normal situation where an error occurs while getting the replicaset revision, so my intention is that the caller be made aware of the problem by printing a warning message. If this happens, then either there is something corrupt about that replicaset, or perhaps a bug in the revision logic somewhere in k8s that causes that situation (should it even occur).

Good point